java/ql/src/Security/CWE/CWE-441/UnsafeContentUriResolution.java

```diff
@@ -0,0 +1,41 @@
+import android.content.ContentResolver;
+import android.net.Uri;
+
+public class Example extends Activity {
+    public void onCreate() {
+        // BAD: Externally-provided URI directly used in content resolution
+        {
+            ContentResolver contentResolver = getContentResolver();
+            Uri uri = (Uri) getIntent().getParcelableExtra("URI_EXTRA");
+            InputStream is = contentResolver.openInputStream(uri);
+            copyToExternalCache(is);
+        }
+        // BAD: input Uri is not normalized, and check can be bypassed with ".." characters
```

Comment:
Suggest using the same capitalization as in the previous comment:

```suggestion
        // BAD: input URI is not normalized, and check can be bypassed with ".." characters
```